beets/importer.py

```diff
@@ -811,55 +811,74 @@ def reimport_metadata(self, lib):
-                    replaced_album.id,
-                    displayable_path(self.album.path)
-                )
+                    "Reimported album {} attribute: ['added'] from album {}, "
+                    "Path: {}",
+                    self.album.id, replaced_album.id,
+                    displayable_path(self.album.path))
+                log.debug(
+                    'Reimported album {} flexible attributes: {} '
+                    'from album {}, Path:{}',
+                    self.album.id, list(preserved_album_attrs.keys()),
+                    replaced_album.id, displayable_path(self.album.path))
 
         for item in self.imported_items():
             dup_items = self.replaced_items[item]
             for dup_item in dup_items:
                 if dup_item.added and dup_item.added != item.added:
                     item.added = dup_item.added
                     log.debug(
-                        'Reimported item added {0} '
-                        'from item {1} for {2}',
-                        item.added,
-                        dup_item.id,
-                        displayable_path(item.path)
-                    )
-                # We exclude certain flexible attributes from the preserving
-                # process since they might have been fetched from MusicBrainz
-                # and been set in beets.autotag.apply_metadata().
-                # discogs_albumid might also have been set but is not a
-                # flexible attribute, thus no exclude is required.
-                if item.get('bandcamp_album_id'):
-                    dup_item.bandcamp_album_id = item.bandcamp_album_id
-                if item.get('spotify_album_id'):
-                    dup_item.spotify_album_id = item.spotify_album_id
-                if item.get('deezer_album_id'):
-                    dup_item.deezer_album_id = item.deezer_album_id
-                if item.get('beatport_album_id'):
-                    dup_item.beatport_album_id = item.beatport_album_id
-                item.update(dup_item._values_flex)
+                        "Reimported item {} attribute: ['added'] "
+                        "from item {}, Path: {}",
+                        item.id, dup_item.id, displayable_path(item.path))
+
+                preserved_item_attrs = dict(dup_item._values_flex)
+                for attr in dont_preserve_item_flex:
+                    if not item.get(attr):
+                        continue
+                    new_val = item.get(attr)
+                    old_val = dup_item.get(attr)
+                    if new_val != old_val:
+                        preserved_item_attrs.pop(attr)
+                        log.debug(
+                            'Not reimported item {} flexible attribute: {}'
+                            'New value: {}, Path: {}',
+                            item.id, attr, new_val,
+                            displayable_path(item.path))
```

Comment:
Since we end up doing more or less the same thing for items and albums, it would be really cool to reduce some of the code duplication that results. Maybe a very small helper function could be the right thing here?